src/main/java/cubicchunks/server/chunkio/IONbtReader.java

```diff
@@ -178,6 +178,9 @@ private static void readEntities(NBTTagCompound nbt, ICubicWorldServer world, Cu
 			}
 
 			entity.addedToChunk = true;
+			entity.chunkCoordX = cube.getX();
```

Comment:
I would add comment eplaining consequences of what bad things may happen if you don't do that.